python/nano/src/bigdl/nano/deps/inference/model.py

```diff
@@ -0,0 +1,123 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import List, Tuple
+from bigdl.nano.pytorch.lightning import LightningModuleFromTorch
+import inspect
+from torch.utils.data import DataLoader
+import torch
+
+
+class AcceleratedLightningModule(LightningModuleFromTorch):
+    def __init__(self, model):
+        super().__init__(model)
+        self.on_init_end()
+
+    def on_init_end(self, *args):
+        pass
+
+    def forward(self, *inputs):
+        inputs = self.on_forward_start(inputs)
+        outputs = self.forward_step(*inputs)
+        return self.on_forward_end(outputs)
+
+    def train(self, mode=True):
+        if mode:
+            raise RuntimeError("This model is not trainable!")
+        super().train(mode)
+
+    def on_forward_start(self, inputs):
+        return inputs
+
+    def forward_step(self, *inputs):
+        return self.model(*inputs)
+
+    def on_forward_end(self, outputs):
+        return outputs
+
+    def get_forward_args(self):
+        return get_forward_args(self)
+
+    @staticmethod
+    def tensors_to_numpy(tensors):
+        np_data = tuple(map(lambda x: x.cpu().numpy(), tensors))
+        return np_data
+
+    @staticmethod
+    def numpy_to_tensors(np_array):
+        tensors = tuple(map(lambda x: torch.from_numpy(x), np_array))
+        if len(tensors) == 1:
+            tensors = tensors[0]
+        return tensors
+
+
+def get_forward_args(model):
+    forward_args = inspect.getfullargspec(model.forward).args[1:]
+    if isinstance(model, LightningModuleFromTorch):
+        # forward param list for compiled model
+        forward_args = get_forward_args(model.model)
+    return forward_args
+
+
+def get_input_example(model: LightningModuleFromTorch, input_sample):
+    if isinstance(input_sample, DataLoader):
+        # TODO: This assumpe the last output is y
+        input_sample = tuple(next(iter(input_sample))[:-1])
+    elif input_sample is None:
+        if getattr(model, "example_input_array", None) is not None:
+            input_sample = model.example_input_array
+        elif model.trainer is not None:
+            for dataloader in [model.test_dataloader(), model.train_dataloader(),
+                               model.val_dataloader()]:
+                if dataloader is not None:
+                    # TODO: This assumpe the last output is y
+                    input_sample = tuple(next(iter(dataloader))[:-1])
+                    break
+            if input_sample is None and model.predict_dataloader():
+                input_sample = tuple(next(iter(model.predict_dataloader())))
+        else:
+            raise RuntimeError("You must specify an input_sample or call `Trainer.fit` "
+                               "on the model first to use `eval_openvino`")
+
+    model.example_input_array = input_sample
+    return input_sample
+
+
+def export(model, input_sample=None, onnx_path="model.onnx", dynamic_axes=True):
```

Comment:
why onnx specific code in this class?

Reply:
This is inference-specific code and no onnx-related import here.  
Here `export` converts a pytorch model to onnx.